### PR TITLE
fcitx5-pinyin-{minecraft,moegirl,zhwiki}: init

### PR DESCRIPTION
**Pull request by Guanran928**
## Description of changes

Adds 3 Fcitx5 dictionary files. 
- https://github.com/oldherl/fcitx5-pinyin-minecraft
- https://github.com/outloudvi/mw2fcitx
- https://github.com/felixonmars/fcitx5-pinyin-zhwiki

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/by-name/fc/fcitx5-pinyin-minecraft/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    runHook postInstall
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Fcitx 5 pinyin dictionary from zh.minecraft.wiki";
+    homepage = "https://github.com/oldherl/fcitx5-pinyin-minecraft";
+    license = lib.licenses.unlicense;
```
Comment (Aleksanaa) [inline pkgs/by-name/fc/fcitx5-pinyin-minecraft/package.nix:31]:
```suggestion
    license = with lib.licenses; [ unlicense cc-by-nc-sa-30 ];
```

See the footer of zh.minecraft.wiki

File: pkgs/by-name/fc/fcitx5-pinyin-moegirl/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    runHook postInstall
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Fcitx 5 pinyin dictionary from zh.moegirl.org.cn";
+    homepage = "https://github.com/outloudvi/mw2fcitx";
+    license = lib.licenses.unlicense;
```
Comment (Aleksanaa) [inline pkgs/by-name/fc/fcitx5-pinyin-moegirl/package.nix:31]:
```suggestion
    license = with lib.licenses; [ unlicense cc-by-nc-sa-30 ];
```

Also see footer

File: pkgs/by-name/fc/fcitx5-pinyin-zhwiki/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    install -Dm644 $src $out/share/fcitx5/pinyin/dictionaries/zhwiki.dict
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Fcitx 5 pinyin dictionary from zh.wikipedia.org";
+    homepage = "https://github.com/felixonmars/fcitx5-pinyin-zhwiki";
+    license = lib.licenses.unlicense;
```
Comment (Aleksanaa) [inline pkgs/by-name/fc/fcitx5-pinyin-zhwiki/package.nix:31]:
```suggestion
    license = with lib.licenses; [ unlicense cc-by-sa-40 ];
```

Also see footer

File: pkgs/by-name/fc/fcitx5-pinyin-zhwiki/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
+{
+  lib,
+  fetchurl,
+  stdenvNoCC,
+}:
+let
+  date = "20240509";
```
Comment (Aleksanaa) [inline pkgs/by-name/fc/fcitx5-pinyin-zhwiki/package.nix:7]: put date inside and reference with `finalAttrs`, so others can override it.

**Review by Aleksanaa — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**